SPECS/mozjs/mozjs.spec

```diff
@@ -0,0 +1,116 @@
+
+%global	major 78
+Summary:       Mozilla's JavaScript engine.
+Name:          mozjs
+Version:       78.10.0
+Release:       2%{?dist}
+Group:         Applications/System
+Vendor:        Microsoft Corporation
+License:       MPLv2.0 and MPLv1.1 and BSD and GPLv2+ and GPLv3+ and LGPLv2+ and AFL and ASL 2.0 and CCO
+URL:           https://developer.mozilla.org/en-US/docs/Mozilla/Projects/SpiderMonkey
+Source0:       https://ftp.mozilla.org/pub/firefox/releases/%{version}esr/source/firefox-%{version}esr.source.tar.xz
+Patch0:        emitter.patch
+Patch1:        emitter_test.patch
+# Build fixes
+Patch2:        init_patch.patch
+Patch3:        spidermonkey_checks_disable.patch
+Distribution:  Mariner
+BuildRequires: which
+BuildRequires: python3-xml
+BuildRequires: python3-libs
+BuildRequires: python3-devel
+BuildRequires: zlib-devel
+BuildRequires: gcc
+BuildRequires: llvm
+BuildRequires: llvm-devel
+BuildRequires: icu-devel
+BuildRequires: rust
+BuildRequires: autoconf213
+Requires:      icu
+Requires:      python3
+Requires:      python3-libs
+Obsoletes:     mozjs60
+Obsoletes:     mozjs68
+Obsoletes:     js
+Provides:      mozjs%{major}
+
+%description
+Mozilla's JavaScript engine includes a just-in-time compiler (JIT) that compiles
+JavaScript to machine code, for a significant speed increase.
+
+%package       devel
+Summary:       mozjs devel
+Group:         Development/Tools
+Requires:      %{name} = %{version}-%{release}
+
+%description   devel
+This contains development tools and libraries for SpiderMonkey.
+
+%prep
+%autosetup -p1 -n firefox-%{version}
+rm -rf modules/zlib
+
+%build
+cd js/src
+%configure \
+    --with-system-icu \
+    --enable-readline \
+    --disable-jemalloc \
+    --disable-tests \
+    --with-system-zlib
+%make_build
+
+%install
+cd js/src
+%make_install
+chmod -x %{buildroot}%{_libdir}/pkgconfig/*.pc
+# remove non required files
+rm %{buildroot}%{_libdir}/libjs_static.ajs
+rm -rf %{buildroot}/usr/src
+find %{buildroot} -name '*.la' -delete
+
+%post
+%ldconfig_scriptlets
+
+%postun
+%ldconfig_scriptlets
+
+
+%files
+%defattr(-,root,root)
+%{_bindir}/js%{major}
+%{_bindir}/js%{major}-config
+%{_libdir}/libmozjs-%{major}.so
```

Comment:
Can we move the `*.so` files to the `devel` package?

Reply:
I think it's fine as is, since this is the actual library file (with the sover being omitted in favor of using the major version in the name).

We could take the approach of artificially adding a sover (like [Fedora does](https://src.fedoraproject.org/rpms/mozjs78/blob/rawhide/f/mozjs78.spec#_241)), but I really don't see what utility that provides to us.

Reply:
This is somewhat suspicious in that case. The Fedora version does provide `libmozjs-[version].so.0.0.0` and `libmozjs-[version].so.0` through the default package, while `libmozjs-68.so` comes from `*-devel`. Either I'm missing something or this package seems misconfigured.

Reply:
I think you may be missing the part in the Feodra spec's install section where they add a fake sover- copied here for clarity:

```bash
mv %{buildroot}%{_libdir}/libmozjs-%{major}.so \
   %{buildroot}%{_libdir}/libmozjs-%{major}.so.0.0.0
ln -s libmozjs-%{major}.so.0.0.0 %{buildroot}%{_libdir}/libmozjs-%{major}.so.0
ln -s libmozjs-%{major}.so.0 %{buildroot}%{_libdir}/libmozjs-%{major}.so
```